packages/core/src/conflict-resolution.ts

```diff
@@ -389,4 +373,26 @@ export class ConflictResolution {
     const handler = this.handlers.get(initialState.type)
     return this.applyLogToState(handler, resetLog)
   }
+
+  /**
+   * Return `CommitData` with commit and JWS envelope, if applicable and not already present.
+   */
+  private async getCommitData(logEntry: LogEntry): Promise<CommitData> {
```

Comment:
How does this look @stbrody? Passing `LogEntry` is not quite necessary right now since `CommitData` is available wherever this method is called but could be useful during further optimizations.

Reply:
If there's a cleaner way to do this let me know (cc @ukstv).

Reply:
Although, if the method took `CommitData` like it did originally, wouldn't it work just the same? `CommitData` extends `LogEntry` and so passing a `LogEntry` structure would just be equivalent to a `CommitData` structure without the `commit` and `envelope` fields (which are optional).

Reply:
Seems like you should just use the *CommitData* type here?
```suggestion
  private async getCommitData(logEntry: CommitData): Promise<CommitData> {
```

Reply:
Yes, that's how it was before. Will revert!